CopilotKit/packages/runtime/src/lib/runtime/__tests__/mcp-tools-utils.test.ts

```diff
@@ -41,238 +40,424 @@ describe("MCP Tools Utils", () => {
 
     it("should extract parameters from schema.parameters.jsonSchema", () => {
       const tool: MCPTool = {
-        description: "Test tool",
+        description: "Test tool with jsonSchema",
         schema: {
           parameters: {
             jsonSchema: {
               properties: {
-                query: { type: "string", description: "Search query" },
+                title: { type: "string", description: "A title parameter" },
+                count: { type: "number", description: "A count parameter" },
+              },
+              required: ["title"],
+            },
+          },
+        },
+        execute: async () => ({}),
+      };
+
+      const result = extractParametersFromSchema(tool);
+      expect(result).toHaveLength(2);
+      expect(result[0]).toEqual({
+        name: "title",
+        type: "string",
+        description: "A title parameter",
+        required: true,
+      });
+      expect(result[1]).toEqual({
+        name: "count",
+        type: "number",
+        description: "A count parameter",
+        required: false,
+      });
+    });
+
+    it("should handle arrays with items", () => {
+      const tool: MCPTool = {
+        description: "Test tool with array parameters",
+        schema: {
+          parameters: {
+            properties: {
+              simpleArray: {
+                type: "array",
+                items: { type: "string" },
+                description: "Array of strings",
+              },
+              objectArray: {
+                type: "array",
+                items: {
+                  type: "object",
+                  properties: {
+                    name: { type: "string" },
+                    value: { type: "number" },
+                  },
+                },
+                description: "Array of objects",
               },
-              required: ["query"],
             },
+            required: ["simpleArray"],
           },
         },
         execute: async () => ({}),
       };
 
-      const parameters = extractParametersFromSchema(tool);
+      const result = extractParametersFromSchema(tool);
+      expect(result).toHaveLength(2);
+      expect(result[0]).toEqual({
+        name: "simpleArray",
+        type: "array<string>",
+        description: "Array of strings",
+        required: true,
+      });
+      expect(result[1]).toEqual({
+        name: "objectArray",
+        type: "array",
+        description: "Array of objects Array of objects with properties: name, value",
```

Comment:
The information of the shape for the nested properties gets lost. Especially when using general names like `value`